src/core/CMakeLists.txt

```diff
@@ -2612,17 +2612,10 @@ if (WITH_PDAL)
 endif()
 
 if (WITH_QTWEBENGINE)
-    if(BUILD_WITH_QT6)
-      target_link_libraries(qgis_core
-        ${QT_VERSION_BASE}::WebEngineCore
-      )
-    else()
-      # QWebEnginePage is in WebEngineWidgets for Qt5
-      target_link_libraries(qgis_core
-        ${QT_VERSION_BASE}::WebEngineCore
-        ${QT_VERSION_BASE}::WebEngineWidgets
-      )
-    endif()
+  target_link_libraries(qgis_core
+    ${QT_VERSION_BASE}::WebEngineCore
+    ${QT_VERSION_BASE}::WebEngineWidgets
```

Comment:
as above -- can we try to avoid the widget import where we can (ie qt 6), and only do it for the gui library?